test/Microsoft.AspNetCore.WebUtilities.Tests/HttpRequestStreamReaderTest.cs

```diff
@@ -242,6 +242,17 @@ public static void StreamDisposed_ExpectedObjectDisposedException(Action<HttpReq
             });
         }
 
+        [Fact]
+        public static void StreamDisposed_ExpectObjectDisposedExceptionAsync()
+        {
+            var httpRequestStreamReader = new HttpRequestStreamReader(new MemoryStream(), Encoding.UTF8, 10, ArrayPool<byte>.Shared, ArrayPool<char>.Shared);
+            httpRequestStreamReader.Dispose();
+
+            Assert.ThrowsAsync<ObjectDisposedException>(async () =>
```

Comment:
The functions below are already async. I don't understand the value in doing this.